lti_consumer/lti_consumer.py

```diff
@@ -991,6 +1008,70 @@ def public_keyset_endpoint(self, request, suffix=''):  # pylint: disable=unused-
             )
         return Response(status=404)
 
+    @XBlock.handler
+    def lti_1p3_access_token(self, request, suffix=''):  # pylint: disable=unused-argument
+        """
+        XBlock handler for creating access tokens for the LTI 1.3 tool.
+
+        This endpoint is only valid when a LTI 1.3 tool is being used.
+
+        Returns:
+            webob.response:
+                Either an access token or error message detailing the failure.
+                All responses are RFC 6749 compliant.
+
+        References:
+            Sucess: https://tools.ietf.org/html/rfc6749#section-4.4.3
+            Failure: https://tools.ietf.org/html/rfc6749#section-5.2
+        """
+        if self.lti_version != "lti_1p3":
+            return Response(status=404)
+        elif request.method != "POST":
+            return Response(status=405)
+
+        lti_consumer = self._get_lti1p3_consumer()
+        try:
+            token = lti_consumer.access_token(
+                dict(parse.parse_qsl(
+                    request.body.decode('utf-8'),
+                    keep_blank_values=True
+                ))
+            )
+            # The returned `token` is compliant with RFC 6749 so we just
+            # need to return a 200 OK response with the token as Json body
+            return Response(token, content_type="application/json")
+
+        # Handle errors and return a proper response
+        # pylint: disable=bare-except
+        except MissingRequiredClaim:
```

Comment:
@viadanna Improved exception handling here. 
I've tried to address all applicable errors from https://tools.ietf.org/html/rfc6749#section-5.2